lib/inc/internal/simple_include_context.hpp

```diff
@@ -6,15 +6,18 @@
 namespace hocon {
 
     class simple_include_context : public config_include_context {
+        using weak_parseable = std::weak_ptr<const parseable>;
     public:
-        simple_include_context(std::shared_ptr<parseable> parseable);
+        // Include context is part of a parseable, so it can only keep a weak pointer
+        // to the owning parseable.
+        simple_include_context(weak_parseable parseable);
```

Comment:
Turns out this is only ever created in the constructor of parseable, and always passed a self reference. So I'm going to avoid this problem by passing in a reference instead. This is one of those places a Core Guidelines checker would be nice, to guarantee no one ever passes a bad reference.